backend/app/services/file.py

```diff
@@ -29,6 +48,101 @@ def _check_if_text_based_file(file: bytes, mime_type: str) -> None:
         raise HTTPException(status.HTTP_400_BAD_REQUEST, f"File with mime type {mime_type} is not text-based")
 
 
+def _is_file_text_based(file: bytes) -> bool:
+    try:
+        file.decode("utf-8")
+        return True
+    except UnicodeDecodeError:
+        return False
+
+
 def _get_blob_url(blob_path: str) -> str:
     quoted_blob_path = quote(blob_path)
     return f"{settings.BLOB_STORAGE_URL}/{quoted_blob_path}"
+
+
+def is_zip(file: UploadFile) -> bool:
+    return file.content_type == "application/zip"
+
+
+async def upload_files_from_zip(zip_file: UploadFile) -> list[str]:
+    tmp_root_dir_name = str(uuid.uuid4())
+    async with unzip_file(zip_file, tmp_root_dir_name) as extracted_files:
+        return await upload_multiple_files(extracted_files, tmp_root_dir_name)
+
+
+@contextlib.asynccontextmanager
+async def unzip_file(file: UploadFile, root_dir_name: str) -> AsyncContextManager[list[Path]]:
```

Comment:
I am not sure, but shouldn't the return type be AsyncGenerator, as the @contextlib.asynccontextmanager documentation says(some_async_generator)?